src/jit/valuenum.cpp

```diff
@@ -3417,22 +3417,40 @@ ValueNum ValueNumStore::EvalUsingMathIdentity(var_types typ, VNFunc func, ValueN
     }
     else // must be a VNF_ function
     {
-        // These identities do not apply for floating point (when x == NaN)
-        //
-        if (VNIsEqual(arg0VN, arg1VN))
+        switch (func)
         {
-            // x <= x == true
-            // x >= x == true
-            if ((func == VNF_LE_UN) || (func == VNF_GE_UN))
-            {
-                resultVN = VNOneForType(typ);
-            }
-            // x < x == false
-            // x > x == false
-            else if ((func == VNF_LT_UN) || (func == VNF_GT_UN))
-            {
-                resultVN = VNZeroForType(typ);
-            }
+            case VNF_LT_UN:
+                // x < 0 == false
+                // x < x == false
+                std::swap(arg0VN, arg1VN);
+                __fallthrough;
+            case VNF_GT_UN:
+                // 0 > x == false
+                // x > x == false
+                if (varTypeIsIntegralOrI(TypeOfVN(arg0VN)) &&
```

Comment:
Might want to mention that this specifically excludes floating point types
Beacuse the 'func' means something completelt different for floating point.
While this is passable in the IL opcodes, the JIT probably should create a different func to use for floating point unordered comparisons.

Reply:
IMO it would be better if JIT has specific float opers not just for comparisons but for all other operations (add, sub etc.) so we don't have to keep checking types. But that's another story.

I'll add the suggested comment. It's easy to forget that _UN also means unordered and think that the integer type check is pointless.